src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.NativeAot.cs

```diff
@@ -361,6 +361,26 @@ public static void PrepareDelegate(Delegate d)
 
             return RuntimeImports.RhNewObject(mt);
         }
+
+        public static unsafe object Box(ref byte target, RuntimeTypeHandle type)
+        {
+            if (type.IsNull)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.type);
+
+            // Compatibility with CoreCLR, throw on a null reference to the unboxed data.
+            if (Unsafe.IsNullRef(ref target))
+                throw new NullReferenceException();
+
+            MethodTable* eeType = type.ToMethodTable();
+
+            if (eeType->IsByRefLike)
```

Comment:
We can move this check under `if (eeType->IsValueType)` to make the non-valuetype case a bit faster for free.